src/Compiler/Driver/ParseAndCheckInputs.fs

```diff
@@ -1961,7 +1961,54 @@ let CheckMultipleInputsUsingGraphMode
 
         partialResults, tcState)
 
+let TryReuseTypecheckingResults (tcConfig: TcConfig) inputs =
+    let tcDataFileName = FileSystem.GetFullPathShim "FSharpTypecheckingData"
+    let nl = Environment.NewLine
+
+    let sourceFiles =
+        inputs
+        |> Seq.toArray
+        |> Array.mapi (fun idx (input: ParsedInput) ->
+            {
+                Idx = idx
+                FileName = input.FileName
+                ParsedInput = input
+            })
+
+    let filePairs = FilePairMap sourceFiles
+
+    let graph =
+        DependencyResolution.mkGraph filePairs sourceFiles
+        |> fst
+        |> Graph.map (fun idx ->
+            let friendlyFileName =
+                sourceFiles[idx]
+                    .FileName.Replace(tcConfig.implicitIncludeDir, "")
+                    .TrimStart([| '\\'; '/' |])
+
+            (idx, friendlyFileName))
+        |> Graph.asString
```

Comment:
Why a string, which on top needed many allocations to be even created?

Reply:
But FileName.Replace and TrimEnd does not - for every single file, on every change, even when there is nothing to reuse - this will add overhead.
(its fine to allocate allocate for essential features, but this one is not essential).

Reply:
Emm alright, let's remove this as well.